### ISSUE TITLE
Schedule is not working as expected with pod-kill sometimes

### ISSUE THREAD

**Issue by iguoyr** [chaos-mesh/chaos-mesh#2387]
## Bug Report

**What version of Kubernetes are you using?**

**What version of Chaos Mesh are you using?**

**What did you do?**

**What did you expect to see?**

pod will be killed at every schedule time

**What did you see instead?**

there is no pod killed after a while (maybe 1-2min)
```
controlplane $ kubectl get -n chaos-sandbox deployments,pods,services -l chaos=blast-here
NAME                        READY   STATUS    RESTARTS   AGE
pod/nginx-79cf599cf-dlzcv   1/1     Running   0          2s
pod/nginx-79cf599cf-j7kdl   1/1     Running   0          64s
pod/nginx-79cf599cf-jtm9r   1/1     Running   0          64s
pod/nginx-79cf599cf-khsgs   1/1     Running   0          64s
pod/nginx-79cf599cf-lmphq   1/1     Running   0          17s
pod/nginx-79cf599cf-lr89c   1/1     Running   0          64s
pod/nginx-79cf599cf-pvhc6   1/1     Running   0          64s
pod/nginx-79cf599cf-xth72   1/1     Running   0          32s
controlplane $ kubectl get -n chaos-sandbox deployments,pods,services -l chaos=blast-here
NAME                        READY   STATUS    RESTARTS   AGE
pod/nginx-79cf599cf-dlzcv   1/1     Running   0          4m6s
pod/nginx-79cf599cf-j7kdl   1/1     Running   0          5m8s
pod/nginx-79cf599cf-jtm9r   1/1     Running   0          5m8s
pod/nginx-79cf599cf-khsgs   1/1     Running   0          5m8s
pod/nginx-79cf599cf-lmphq   1/1     Running   0          4m21s
pod/nginx-79cf599cf-lr89c   1/1     Running   0          5m8s
pod/nginx-79cf599cf-pvhc6   1/1     Running   0          5m8s
pod/nginx-79cf599cf-xth72   1/1     Running   0          4m36s
```

schedule detail
```
controlplane $ kubectl get Schedule -n chaos-mesh
NAME               AGE
pod-kill-example   5m3s
controlplane $ kubectl get Schedule -n chaos-mesh pod-kill-example -o yaml
apiVersion: chaos-mesh.org/v1alpha1
kind: Schedule
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"chaos-mesh.org/v1alpha1","kind":"Schedule","metadata":{"annotations":{},"name":"pod-kill-example","namespace":"chaos-mesh"},"spec":{"concurrencyPolicy":"Forbid","historyLimit":5,"podChaos":{"action":"pod-kill","mode":"one","selector":{"labelSelectors":{"chaos":"blast-here"},"namespaces":["chaos-sandbox"]}},"schedule":"@every 15s","type":"PodChaos"}}
  creationTimestamp: "2021-10-13T03:27:06Z"
  generation: 26
  name: pod-kill-example
  namespace: chaos-mesh
  resourceVersion: "1612"
  uid: a56484a4-fb68-4f10-8b47-b6d696d0c51e
spec:
  concurrencyPolicy: Forbid
  historyLimit: 5
  podChaos:
    action: pod-kill
    gracePeriod: 0
    mode: one
    selector:
      labelSelectors:
        chaos: blast-here
      namespaces:
      - chaos-sandbox
  schedule: '@every 15s'
  startingDeadlineSeconds: null
  type: PodChaos
status:
  active:
  - apiVersion: chaos-mesh.org/v1alpha1
    kind: PodChaos
    name: pod-kill-example-h97bg
    namespace: chaos-mesh
    resourceVersion: "1605"
    uid: 0e479d35-f75c-4a31-9f02-73a4d506cabf
  - apiVersion: chaos-mesh.org/v1alpha1
    kind: PodChaos
    name: pod-kill-example-jjws9
    namespace: chaos-mesh
    resourceVersion: "1485"
    uid: ad16f021-a640-4413-97ab-4a6f904ebaf0
  - apiVersion: chaos-mesh.org/v1alpha1
    kind: PodChaos
    name: pod-kill-example-t7gq9
    namespace: chaos-mesh
    resourceVersion: "1554"
    uid: 5af056ff-b792-4836-809a-abb321b2af65
  time: "2021-10-13T03:27:51Z"
```

chaos-controller-manager log
```
2021-10-13T03:33:19.202Z        INFO    records iterating record        {"record": {"id":"chaos-sandbox/nginx-79cf599cf-hc88r","selectorKey":".","phase":"Not Injected"}, "desiredPhase": "Run"}
2021-10-13T03:33:19.202Z        INFO    records apply chaos     {"id": "chaos-sandbox/nginx-79cf599cf-hc88r"}
2021-10-13T03:33:19.202Z        ERROR   records fail to apply chaos     {"error": "Pod \"nginx-79cf599cf-hc88r\" not found"}
github.com/go-logr/zapr.(*zapLogger).Error
        /tmp/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/chaos-mesh/chaos-mesh/controllers/common.(*Reconciler).Reconcile
        /mnt/controllers/common/controller.go:170
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:256
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
        /tmp/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
        /tmp/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
        /tmp/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:88
```

**Output of chaosctl**

**Comment by YangKeao** [comment]
While investigating this problem, I got some wired log:

```
2021-10-21T06:09:41.199Z        INFO    records updating records        {"records": [{"id":"default/nginx-79cf599cf-frwmj","selectorKey":".","phase":"Injected"}]}
2021-10-21T06:09:41.202Z        DEBUG   controller-runtime.webhook.webhooks     received request        {"webhook": "/mutate-chaos-mesh-org-v1alpha1-podchaos", "UID": "a3f47ad5-3b33-484d-bde4-bfc93accb223", "kind": "chaos-mesh.org/v1alpha1, Kind=PodChaos", "resource": {"group":"chaos-mesh.org","version":"v1alpha1","resource":"podchaos"}}
2021-10-21T06:09:41.202Z        INFO    podchaos-resource       default {"name": "schedule-pod-kill-example-wk2f7"}
2021-10-21T06:09:41.203Z        DEBUG   controller-runtime.webhook.webhooks     wrote response  {"webhook": "/mutate-chaos-mesh-org-v1alpha1-podchaos", "UID": "a3f47ad5-3b33-484d-bde4-bfc93accb223", "allowed": true, "result": {}, "resultError": "got runtime.Object without object metadata: &Status{ListMeta:ListMeta{SelfLink:,ResourceVersion:,Continue:,RemainingItemCount:nil,},Status:,Message:,Reason:,Details:nil,Code:200,}"}
2021-10-21T06:09:41.206Z        DEBUG   controller-runtime.webhook.webhooks     received request        {"webhook": "/validate-chaos-mesh-org-v1alpha1-podchaos", "UID": "acd70a45-8c82-4d85-81e6-07ec598b12b3", "kind": "chaos-mesh.org/v1alpha1, Kind=PodChaos", "resource": {"group":"chaos-mesh.org","version":"v1alpha1","resource":"podchaos"}}
2021-10-21T06:09:41.206Z        INFO    podchaos-resource       validate update {"name": "schedule-pod-kill-example-wk2f7"}
2021-10-21T06:09:41.207Z        DEBUG   controller-runtime.webhook.webhooks     wrote response  {"webhook": "/validate-chaos-mesh-org-v1alpha1-podchaos", "UID": "acd70a45-8c82-4d85-81e6-07ec598b12b3", "allowed": true, "result": {}, "resultError": "got runtime.Object without object metadata: &Status{ListMeta:ListMeta{SelfLink:,ResourceVersion:,Continue:,RemainingItemCount:nil,},Status:,Message:,Reason:,Details:nil,Code:200,}"}
2021-10-21T06:09:41.207Z        DEBUG   controller-runtime.webhook.webhooks     received request        {"webhook": "/validate-auth", "UID": "e48fb65d-7043-4c45-b625-5312056039b2", "kind": "chaos-mesh.org/v1alpha1, Kind=PodChaos", "resource": {"group":"chaos-mesh.org","version":"v1alpha1","resource":"podchaos"}}
2021-10-21T06:09:41.209Z        INFO    validate-auth   user have the privileges on namespace, auth validate passed     {"user": "system:serviceaccount:chaos-mesh:chaos-controller-manager", "groups": ["system:serviceaccounts", "system:serviceaccounts:chaos-mesh", "system:authenticated"], "namespace": {"default":{}}}
2021-10-21T06:09:41.209Z        DEBUG   controller-runtime.webhook.webhooks     wrote response  {"webhook": "/validate-auth", "UID": "e48fb65d-7043-4c45-b625-5312056039b2", "allowed": true, "result": {}, "resultError": "got runtime.Object without object metadata: &Status{ListMeta:ListMeta{SelfLink:,ResourceVersion:,Continue:,RemainingItemCount:nil,},Status:,Message:,Reason:,Details:nil,Code:200,}"}
2021-10-21T06:09:41.210Z        DEBUG   controller-runtime.controller   Successfully Reconciled {"controller": "podchaos-condition", "request": "default/schedule-pod-kill-example-wk2f7"}
2021-10-21T06:09:41.211Z        INFO    desiredphase    modify desiredPhase     {"desiredPhase": "Run"}
2021-10-21T06:09:41.211Z        DEBUG   controller-runtime.controller   Successfully Reconciled {"controller": "podchaos-desiredphase", "request": "default/schedule-pod-kill-example-wk2f7"}
2021-10-21T06:09:41.211Z        DEBUG   controller-runtime.controller   Successfully Reconciled {"controller": "schedule-gc", "request": "default/schedule-pod-kill-example"}
2021-10-21T06:09:41.211Z        INFO    schedule-active updating active {"active": [{"kind":"PodChaos","namespace":"default","name":"schedule-pod-kill-example-wk2f7","uid":"6e4ba4c5-6637-418f-994b-726a9c9d062b","apiVersion":"chaos-mesh.org/v1alpha1","resourceVersion":"1318"}]}
2021-10-21T06:09:41.211Z        DEBUG   controller-runtime.controller   Successfully Reconciled {"controller": "podchaos-condition", "request": "default/schedule-pod-kill-example-wk2f7"}
2021-10-21T06:09:41.214Z        DEBUG   controller-runtime.webhook.webhooks     received request        {"webhook": "/mutate-chaos-mesh-org-v1alpha1-podchaos", "UID": "02c0d6c4-3d1b-4596-9ca8-04826d71f1ef", "kind": "chaos-mesh.org/v1alpha1, Kind=PodChaos", "resource": {"group":"chaos-mesh.org","version":"v1alpha1","resource":"podchaos"}}
2021-10-21T06:09:41.214Z        INFO    podchaos-resource       default {"name": "schedule-pod-kill-example-wk2f7"}
2021-10-21T06:09:41.214Z        DEBUG   controller-runtime.webhook.webhooks     wrote response  {"webhook": "/mutate-chaos-mesh-org-v1alpha1-podchaos", "UID": "02c0d6c4-3d1b-4596-9ca8-04826d71f1ef", "allowed": true, "result": {}, "resultError": "got runtime.Object without object metadata: &Status{ListMeta:ListMeta{SelfLink:,ResourceVersion:,Continue:,RemainingItemCount:nil,},Status:,Message:,Reason:,Details:nil,Code:200,}"}
2021-10-21T06:09:41.218Z        DEBUG   controller-runtime.webhook.webhooks     received request        {"webhook": "/validate-auth", "UID": "a66c0fa5-e4e3-429e-a811-93a0d74041d4", "kind": "chaos-mesh.org/v1alpha1, Kind=Schedule", "resource": {"group":"chaos-mesh.org","version":"v1alpha1","resource":"schedules"}}
2021-10-21T06:09:41.218Z        DEBUG   controller-runtime.webhook.webhooks     wrote response  {"webhook": "/validate-auth", "UID": "a66c0fa5-e4e3-429e-a811-93a0d74041d4", "allowed": true, "result": {}, "resultError": "got runtime.Object without object metadata: &Status{ListMeta:ListMeta{SelfLink:,ResourceVersion:,Continue:,RemainingItemCount:nil,},Status:,Message:,Reason:skip the RBAC check for type Schedule,Details:nil,Code:200,}"}
2021-10-21T06:09:41.218Z        DEBUG   controller-runtime.webhook.webhooks     received request        {"webhook": "/validate-chaos-mesh-org-v1alpha1-schedule", "UID": "5981995a-7bfe-4e8f-9f6f-1474eb5c3e10", "kind": "chaos-mesh.org/v1alpha1, Kind=Schedule", "resource": {"group":"chaos-mesh.org","version":"v1alpha1","resource":"schedules"}}
2021-10-21T06:09:41.218Z        INFO    schedule-resource       validate update {"name": "schedule-pod-kill-example"}
2021-10-21T06:09:41.218Z        DEBUG   controller-runtime.webhook.webhooks     wrote response  {"webhook": "/validate-chaos-mesh-org-v1alpha1-schedule", "UID": "5981995a-7bfe-4e8f-9f6f-1474eb5c3e10", "allowed": true, "result": {}, "resultError": "got runtime.Object without object metadata: &Status{ListMeta:ListMeta{SelfLink:,ResourceVersion:,Continue:,RemainingItemCount:nil,},Status:,Message:,Reason:,Details:nil,Code:200,}"}
2021-10-21T06:09:41.225Z        INFO    records updating records        {"records": [{"id":"default/nginx-79cf599cf-frwmj","selectorKey":".","phase":"Not Injected"}]}
```

The two iteration of `records-controller` gets different records. The only possible reason is that the records are modified inside the "retry".

**Comment by YangKeao** [comment]
This issue should have been fixed by #2465
